source/BadgeOverlaySupport.js

```diff
@@ -0,0 +1,220 @@
+(function (enyo, scope) {
+	/**
+	* {@link moon.BadgeOverlaySupport} is a [mixin]{@glossary mixin} that may be applied
+	* to any {@link moon.DataList} or {@link moon.DataGridList} item to provide an overlay
+	* that is activated when the list contains items of type Video/Audio/Image.
+	*
+	* The BadgeOverlaySupport purpose is to show an icon badge on top of item to indicate
+	* the type of item. Like for Video/Audio it will show a default play icon and for Image
+	* there is nothing by default.
+	*
+	* The item has to define a
+	* [badgeType]{@link moon.BadgeOverlaySupport#badgeType} MIME type of this
+	* item. Possible values are 'video', 'audio' , 'image'. Defaulted as 'image'.
+	*
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	* ```
+	*
+	* The item may define a
+	* [badgeSrc]{@link moon.BadgeOverlaySupport#badgeSrc} URL to
+	* override the default icon.
+	*
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video', badgeSrc: 'assets/my-icon.png',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	* ```
+	*
+	* The item may define a
+	* [position]{@link moon.BadgeOverlaySupport#position} to
+	* specify position of badge. Possible values are 'top' or 'bottom'. Defaulted as 'bottom'.
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video', position: 'top',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	*
+	* @mixin moon.BadgeOverlaySupport
+	* @protected
+	*/
+	moon.BadgeOverlaySupport = {
+
+		/**
+		* @private
+		*/
+		name: 'moon.BadgeOverlaySupport',
+
+		/**
+		* @private
+		*/
+		classes: 'moon-badge-overlay-support',
+
+		/**
+		* Identifies the MIME type of badge.
+		* Possible values are "image", "audio" or "video".
+		* Defaulted as 'image'.
+		*
+		* @name moon.BadgeOverlaySupport#badgeType
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeType: null,
+
+		/**
+		* Font icon to be used in place of default icon.
+		*
+		* @name moon.BadgeOverlaySupport#badgeIcon
+		* @type {String}
```

Comment:
For all three of these public properties, the documented default value (`undefined`) doesn't match the actual default value (`null`).